awx/ui/client/src/scheduler/schedulerEdit.controller.js

```diff
@@ -283,7 +284,23 @@ function($filter, $state, $stateParams, Wait, $scope, moment,
                             currentValues: data
                         });
 
-                        prompts.credentials.value = scheduleCredentials.results.length > 0 ? scheduleCredentials.results : prompts.credentials.value;
+                        let defaultCredsWithoutOverrides = [];
+
+                        prompts.credentials.value.forEach((defaultCred) => {
+                            let typeMatches = false;
+                            scheduleCredentials.forEach((scheduleCred) => {
+                                if(defaultCred.credential_type === scheduleCred.credential_type) {
+                                    if((!defaultCred.vault_id && !scheduleCred.inputs.vault_id) || (defaultCred.vault_id && scheduleCred.inputs.vault_id && defaultCred.vault_id === scheduleCred.inputs.vault_id)) {
```

Comment:
Here, there are a lot of conditions that are hard to follow. What do you think about breaking this out into another method?